Add UserCredentialsStrategy (#1558)
hikari-py/hikari: changes/1558.feature.md

> Add `UserCredentialsStrategy` to `hikari\impl\rest.py` similar to `ClientCredentialsStrategy` to allow simple OAuth2
> token generation and refreshing

Reviewer: ```suggestion
Add `OAuthCredentialsStrategy` to `hikari.impl.rest` for OAuth2 flow tokens.
```
